app/src/main/java/rick_and_morty/ui/episodes/composables/EpisodesList.kt

```diff
@@ -1,77 +1,84 @@
-import androidx.compose.material3.Text
 import androidx.compose.runtime.Composable
 import androidx.compose.runtime.LaunchedEffect
 import androidx.compose.runtime.collectAsState
 import androidx.compose.runtime.derivedStateOf
 import androidx.compose.runtime.getValue
 import androidx.compose.runtime.remember
+import androidx.compose.ui.Alignment
 import androidx.compose.ui.Modifier
 import androidx.compose.ui.unit.dp
 import androidx.lifecycle.viewmodel.compose.viewModel
-import androidx.navigation.NavController
-import rick_and_morty.BottomNavigationBar
-import rick_and_morty.eventbus.EventBus
-import rick_and_morty.ui.characters.CharactersViewModel
 import rick_and_morty.ui.episodes.EpisodesViewModel
 import rick_and_morty.ui.widgets.CircularProgressBar
 import rick_and_morty.ui.widgets.RickAndMortyErrorDialog
 
+@OptIn(ExperimentalMaterialApi::class)
 @Composable
 fun EpisodeList(
         episodesViewModel: EpisodesViewModel = viewModel(modelClass = EpisodesViewModel::class.java),
 ) {
 
         val episodes = episodesViewModel.episodes.collectAsState().value
         val scrollListState = rememberLazyListState()
-        if (episodes.isLoading)
-        {
-                CircularProgressBar()
-        }
-        when {
-                episodes.isFailure -> RickAndMortyErrorDialog(episodes.failure.toString())
-                else -> {
-                        LazyColumn (
-                                modifier = Modifier.background(MaterialTheme.colorScheme.primary),
-                                verticalArrangement = Arrangement.spacedBy(5.dp),
-                                state = scrollListState
-                        ) {
-                                items(episodes.episodeResults) { episode ->
-                                        EpisodesRow(
-                                                episodeResultsDto = episode,
-                                          )
-                                }
-                        }
-                        val userIsAtBottom by remember{
-                                derivedStateOf {
-                                        val layoutInfo = scrollListState.layoutInfo
-                                        val visibleItemsInfo = layoutInfo.visibleItemsInfo
-                                        if (layoutInfo.totalItemsCount == 0) {
-                                                false
-                                        } else {
-                                                val lastVisibleItem = visibleItemsInfo.last()
-                                                val viewportHeight = layoutInfo.viewportEndOffset + layoutInfo.viewportStartOffset
-                                                (lastVisibleItem.index + 1 == layoutInfo.totalItemsCount &&
-                                                        lastVisibleItem.offset + lastVisibleItem.size <= viewportHeight)
+
+        val refreshing = episodes.isLoading
```

Comment:
We can save the effort of declaring a variable here. `episodes.isLoading` can also be called in the respective places in the code.

Reply:
Done